src/commands/addons/config.js

```diff
@@ -151,52 +128,44 @@ class AddonsConfigCommand extends Command {
         return false
       }
 
-      await update(
-        {
-          addonName,
-          currentConfig,
-          newConfig,
-          settings: {
-            siteId,
-            instanceId: currentAddon.id,
-            addon: addonName,
-            config: newConfig,
-          },
-          accessToken,
-          error: this.error,
-        },
-        this.log
-      )
+      await update({
+        addonName,
+        currentConfig,
+        newConfig,
+        siteId,
+        instanceId: addon.id,
+        api,
+        error: this.error,
+        log: this.log,
+      })
     }
   }
 }
 
-async function update({ addonName, currentConfig, newConfig, settings, accessToken, error }, logger) {
+async function update({ addonName, currentConfig, newConfig, siteId, instanceId, api, error, log }) {
   const codeDiff = diffValues(currentConfig, newConfig)
   if (!codeDiff) {
-    logger('No changes, exiting early')
+    log('No changes, exiting early')
     return false
   }
-  logger()
+  log()
   const msg = `Updating ${addonName} add-on config values...`
-  logger(`${chalk.white.bold(msg)}`)
-  logger()
-  logger(`${codeDiff}\n`)
-  logger()
+  log(`${chalk.white.bold(msg)}`)
+  log()
+  log(`${codeDiff}\n`)
+  log()
 
-  let updateAddonResponse
   try {
-    // TODO update updateAddon to https://open-api.netlify.com/#operation/updateServiceInstance
-    updateAddonResponse = await updateAddon(settings, accessToken)
+    await api.updateServiceInstance({
+      siteId,
+      addon: addonName,
+      instanceId,
+      body: { config: newConfig },
+    })
+    log(`Add-on "${addonName}" successfully updated`)
   } catch (e) {
     error(e.message)
   }
-  if (updateAddonResponse.code === 404) {
```

Comment:
Should this `404` log message be kept?

Reply:
Similar to other `js-client` methods - a 404 status code will throw an error which will be handled in the `try/catch`

Reply:
:+1: